src/py_avro_schema/_schemas.py

```diff
@@ -443,35 +445,61 @@ def data(self, names: NamesType) -> JSONStr:
 
 class DecimalSchema(Schema):
     """
-    An Avro bytes, logical decimal schema for a Python decimal.Decimal class
+    An Avro bytes, logical decimal schema for a Python :class:`decimal.Decimal`
 
-    For this to work, users must annotate variables with :class:`avro_schema.DecimalType` (standard lib
-    :class:`decimal.Decimal` can be used for values) like so::
+    For this to work, users must annotate variables with like so::
 
        >>> import decimal
-       >>> import py_avro_schema
-       >>> my_decimal: py_avro_schema.DecimalType[4, 2] = decimal.Decimal("12.34")
+       >>> from typing import Annotated
+       >>> my_decimal: Annotated[decimal.Decimal, (4, 2)] = decimal.Decimal("12.34")
     """
 
     @classmethod
     def handles_type(cls, py_type: Type) -> bool:
         """Whether this schema class can represent a given Python class"""
+        try:
+            cls._size(py_type)  # If we can find precision and scale, we are good
```

Comment:
does the avro spec really require this? Can't one use decimals without precision / scale in avro? 🤔

Reply:
Good question, let's check.

I was going on the basis that to fully define a decimal, mathematically, you will need a value for scale and precision. But maybe there are some implicit defaults, e.g. zero scale.

Reply:
From https://avro.apache.org/docs/1.11.1/specification/#decimal

"scale, a JSON integer representing the scale (optional). If not specified the scale is 0."

"Precision must be a positive integer greater than zero"

There are some more restrictions.

The question though is whether there is a **clean** way to omit the scale from the Python (annotated) type hint.

Reply:
Options:

1. `Annotated[decimal.Decimal, DecimalMeta(precision=4, scale=2)` where `DecimalMeta` is a dataclass which defaults scale to zero if not specified.
2. `Annotated[decimal.Decimal, (4, 2)]`, using `Annotated[decimal.Decimal, (4, None)]` to default  scale to zero.

Other options?

Reply:
I do like the first option as it would allow to set it as any arg and not strictly the second arg of Annotated. It's easier to iterate through and find exactly the arg needed. Seems reasonable as more and more libs use this mechanism.

Reply:
yeah, good point about the arg position

Reply:
The only downside is that it will again require a py-avro-schema specific "type". That might simply the price to pay for a robust type hint. I couldnt find anyone else having defined a decimal meta type struct...

Reply:
An option would be to get this from field info of the class itself using dataclasse.Field.metadata (py-avro-schema can choose how this metadata looks it is specifically for third party extension) and in pydantic from FieldInfo of a decimal. If there is non set some sane default?